cmd/monaco/integrationtest/v2/references_test.go

```diff
@@ -0,0 +1,245 @@
+
+	envs := []string{"classic_env", "platform_env"}
+
+	tests := []struct {
+		project      string
+		downloadOpts []string
+		validate     func(t *testing.T, ctx TestContext, confsPerType project.ConfigsPerType)
+	}{
+		{
+			project: "classic-apis",
+			downloadOpts: []string{
+				"-a", "alerting-profile,notification,management-zone",
+			},
+			validate: func(t *testing.T, ctx TestContext, confsPerType project.ConfigsPerType) {
+				managementZone := findConfig(t, confsPerType, "management-zone", "zone_"+ctx.suffix)
+				profile := findConfig(t, confsPerType, "alerting-profile", "profile_"+ctx.suffix)
+				notification := findConfig(t, confsPerType, "notification", "notification_"+ctx.suffix)
+
+				assertRefParamFromTo(t, profile, managementZone)
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+		{
+			project: "settings",
+			downloadOpts: []string{
+				"-s", "builtin:problem.notifications,builtin:management-zones,builtin:alerting.profile",
+			},
+			validate: func(t *testing.T, ctx TestContext, confsPerType project.ConfigsPerType) {
+				managementZone := findSetting(t, confsPerType, "builtin:management-zones", "zone_"+ctx.suffix, "name")
+				profile := findSetting(t, confsPerType, "builtin:alerting.profile", "profile_"+ctx.suffix, "name")
+				notification := findSetting(t, confsPerType, "builtin:problem.notifications", "notification_"+ctx.suffix, "displayName")
+
+				assertRefParamFromTo(t, profile, managementZone)
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+		{
+			project: "settings-with-classic-mngt-zone",
+			downloadOpts: []string{
+				"-a", "management-zone",
+				"-s", "builtin:problem.notifications,builtin:alerting.profile",
+			},
+			validate: func(t *testing.T, ctx TestContext, confsPerType project.ConfigsPerType) {
+				managementZone := findConfig(t, confsPerType, "management-zone", "zone_"+ctx.suffix)
+				profile := findSetting(t, confsPerType, "builtin:alerting.profile", "profile_"+ctx.suffix, "name")
+				notification := findSetting(t, confsPerType, "builtin:problem.notifications", "notification_"+ctx.suffix, "displayName")
+
+				assertRefParamFromTo(t, profile, managementZone)
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+		{
+			project: "config-with-mngt-zone",
+			downloadOpts: []string{
+				"-a", "notification,alerting-profile",
+				"-s", "builtin:management-zones",
+			},
+			validate: func(t *testing.T, ctx TestContext, confsPerType project.ConfigsPerType) {
+				managementZone := findSetting(t, confsPerType, "builtin:management-zones", "zone_"+ctx.suffix, "name")
+				profile := findConfig(t, confsPerType, "alerting-profile", "profile_"+ctx.suffix)
+				notification := findConfig(t, confsPerType, "notification", "notification_"+ctx.suffix)
+
+				assertRefParamFromTo(t, profile, managementZone)
+				assertRefParamFromTo(t, notification, profile)
+			},
+		},
+	}
+
+	for _, env := range envs {
+		for _, tc := range tests {
```

Comment:
nit: any reason to deviate from the usual `tt` naming of the test case var? (other than `tc` actually sounding more correct, just off idiom